ui/packages/catalog-ui-search/src/main/webapp/component/builder/builder.view.js

```diff
@@ -0,0 +1,214 @@
+    });
+
+    return Marionette.LayoutView.extend({
+            template,
+            tagName: CustomElements.register('builder'),
+            modelEvents: {
+                'change:metacard': 'handleMetacard'
+            },
+            events: {
+                'click .builder-edit': 'edit',
+                'click .builder-save': 'save',
+                'click .builder-cancel': 'cancel'
+            },
+            regions: {
+                builderProperties: '> .builder-properties',
+                builderAvailableType: '> .builder-select-available-type > .builder-select-available-type-dropdown'
+            },
+            initialize(options) {
+
+                if (!availableTypes) {
+                    const loadingview = new LoadingView();
+                    ajaxCall.then(() => {
+                        loadingview.remove();
+                        this.model.set('availableTypes', availableTypes);
+                        this.handleAvailableTypes();
+                    });
+                } else {
+                    this.model.set('availableTypes', availableTypes);
+                }
+
+            },
+            isSingleAvailableType() {
+                const availableTypes = this.model.get('availableTypes');
+                return availableTypes && availableTypes.availabletypes && availableTypes.availabletypes.length === 1;
+            },
+            isMultipleAvailableTypes() {
+                const availableTypes = this.model.get('availableTypes');
+                return availableTypes && availableTypes.availabletypes && availableTypes.availabletypes.length > 1;
+            },
+            showMetacardTypeSelection() {
+                const enums = this.model.get('availableTypes').availabletypes.map((availableType) => ({ label: availableType.metacardType, value: availableType.metacardType }));
+
+                const availableTypesModel = new Property({
+                    label: "Select An Available Metacard Type",
+                    value: [this.model.get('availableTypes').availabletypes[0].metacardType],
+                    enum: enums,
+                    id: "Select Metacard Type"
+                });
+
+                this.builderAvailableType.show(new PropertyView({
+                    model: availableTypesModel
+                }));
+
+                this.builderAvailableType.currentView.turnOnEditing();
+
+                this.listenTo(availableTypesModel, 'change:value', this.handleSelectedAvailableType);
+
+                this.$el.addClass('is-selecting-available-types');
+            },
+            handleSystemTypes() {
+
+                metacardDefinitions.metacardDefinitions.reduce((accumulator, currentValue) => {
+                    console.log(currentValue);
+                }, { availabletypes: [] })
+
+                const mds = metacardDefinitions.metacardDefinitions;
+
+                const allTypes = Object.keys(metacardDefinitions.metacardDefinitions)
```

Comment:
✏️ We can use the variable defined above.

Reply:
done